js/src/settings/reconnectionLock.js

```diff
@@ -0,0 +1,96 @@
+/**
```

Comment:
Nice workaround. 

💅  I don't really see the unlock/lock Reconnection as a good naming IMHO. Seems for me more declarative "disableMenuItems" or "enableMenuItems" at the end is what we are doing right?

Reply:
It doesn't only disable the menu links but also locks a link as the available one and tries to redirect the browser to that one when needed. The primary purpose is a bit closer to locking the re-connecting page as the only available page.